test/framework/status.go

```diff
@@ -120,3 +121,39 @@ func (f *Framework) WaitForResourceAvailable(ctx context.Context, getResourceSta
 
 	return nil
 }
+
+// WaitForConfigResourceAcceptedCondition waits for a configuration resource (serviceMonitor, podMonitor, scrapeConfig and probes) to report their status.
+// If the status isn't available within the given timeout, it returns an error.
+func (f *Framework) WaitForConfigResourceAcceptedCondition(ctx context.Context, getConfigResourceStatus func(context.Context) ([]monitoringv1.WorkloadBinding, error), workload metav1.Object, resource string, acceptedStatus monitoringv1.ConditionStatus, timeout time.Duration) error {
+	var pollErr error
+	var bindings []monitoringv1.WorkloadBinding
+	if err := wait.PollUntilContextTimeout(ctx, 5*time.Second, timeout, false, func(ctx context.Context) (bool, error) {
+		bindings, pollErr = getConfigResourceStatus(ctx)
+		if pollErr != nil {
+			return false, nil
+		}
+
+		if len(bindings) == 0 || len(bindings[0].Conditions) == 0 {
```

Comment:
It would be better to loop over the bindings, identify the one matching the workload and check the Accepted condition.

````
for _, binding := range bindings {
    if binding.Namespace != workload.GetNamespace() {
        continue
    }
    ...
}
````